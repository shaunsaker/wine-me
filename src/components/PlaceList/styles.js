import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: {},
  contentContainer: {},
  itemContainer: {
    paddingHorizontal: styleConstants.dimensions.spacing.horizontal,
    paddingBottom: styleConstants.dimensions.spacing.vertical,
  },
});

export default styles;
