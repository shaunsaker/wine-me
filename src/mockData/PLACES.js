import PLACE from './PLACE';

const PLACES = { 1: PLACE };

export default PLACES;
