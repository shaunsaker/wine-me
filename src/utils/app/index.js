import createError from './createError';
import createMapsLinkFromCoordinates from './createMapsLinkFromCoordinates';
import getDistanceBetweenCoordinates from './getDistanceBetweenCoordinates';
import getGooglePlacesPhotoURI from './getGooglePlacesPhotoURI';
import log from './log';
import navigate from './navigate';
import prepareNextAction from './prepareNextAction';

// All app-specific utils go here
const app = {
  createError,
  createMapsLinkFromCoordinates,
  getDistanceBetweenCoordinates,
  getGooglePlacesPhotoURI,
  log,
  navigate,
  prepareNextAction,
};

export default app;
