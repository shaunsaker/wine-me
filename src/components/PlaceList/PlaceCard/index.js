import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import utils from '../../../utils';
import config from '../../../config';

import PlaceCard from './PlaceCard';

export class PlaceCardContainer extends React.Component {
  constructor(props) {
    super(props);

    this.onPress = this.onPress.bind(this);
    this.navigate = this.navigate.bind(this);

    this.state = {};
  }

  static propTypes = {
    place: PropTypes.shape({
      name: PropTypes.string,
      location: PropTypes.shape({
        lat: PropTypes.number,
        lng: PropTypes.number,
      }),
      photoReferences: PropTypes.arrayOf(PropTypes.string),
    }).isRequired,

    // From store
    deviceLocation: PropTypes.shape({
      latitude: PropTypes.number,
      longitude: PropTypes.number,
    }).isRequired,
    userCheckIns: PropTypes.shape({}).isRequired,
  };

  static defaultProps = {};

  onPress() {
    const { place } = this.props;

    this.navigate('place', { place });
  }

  navigate(page, props) {
    utils.app.navigate(page, props);
  }

  render() {
    const { place, deviceLocation, userCheckIns } = this.props;

    // Get the google places photo uri from the photo reference
    const photoReference = place.photoReferences && place.photoReferences[0];
    const uri =
      photoReference &&
      utils.app.getGooglePlacesPhotoURI(
        photoReference,
        config.googlePlaces.maxImageHeight,
        config.googlePlaces.apiKey,
      );
    const imageSource = uri && {
      uri,
    };
    const title = place.name;

    // Get the relative distance between the place and the device location (if it's available)
    let parsedDeviceLocation;
    let relativeDistance;
    let label;

    if (deviceLocation.latitude) {
      parsedDeviceLocation = {
        lat: deviceLocation.latitude,
        lng: deviceLocation.longitude,
      };
      relativeDistance = Math.round(
        utils.app.getDistanceBetweenCoordinates(place.location, parsedDeviceLocation),
      );

      label = `${relativeDistance} km from you`;
    }

    // Check if the user has checked into the place
    const hasCheckedIn = utils.objects
      .convertObjectToArray(userCheckIns)
      .filter((checkIn) => checkIn.place_id === place.id).length
      ? true
      : null;

    return (
      <PlaceCard
        imageSource={imageSource}
        title={title}
        label={label}
        handlePress={this.onPress}
        hasCheckedIn={hasCheckedIn}
        testID="placeCardContainer"
      />
    );
  }
}

function mapStateToProps(state) {
  return {
    deviceLocation: state.appState.deviceLocation,
    userCheckIns: state.appData.userCheckIns,
  };
}

export default connect(mapStateToProps)(PlaceCardContainer);
