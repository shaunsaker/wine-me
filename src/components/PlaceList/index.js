import React from 'react';
import PropTypes from 'prop-types';
import { View, FlatList } from 'react-native';

import styles from './styles';

import PlaceCard from './PlaceCard';

export default class PlaceList extends React.Component {
  constructor(props) {
    super(props);

    this.renderItem = this.renderItem.bind(this);
  }

  static propTypes = {
    data: PropTypes.arrayOf(PropTypes.shape({})),
  };

  static defaultProps = {};

  renderItem({ item }) {
    return (
      <View style={styles.itemContainer}>
        <PlaceCard place={item} />
      </View>
    );
  }

  render() {
    const { data } = this.props;

    return (
      <FlatList
        keyExtractor={({ id }) => id}
        data={data}
        renderItem={this.renderItem}
        style={styles.container}
        contentContainerStyle={styles.contentContainer}
        initialNumToRender={3}
      />
    );
  }
}
