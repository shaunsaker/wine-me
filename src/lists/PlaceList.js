import React from "react";
import { View, FlatList, StyleSheet } from "react-native";
import PropTypes from "prop-types";

import utilities from "../utilities";
import styleConstants from "../assets/styleConstants";

import PlaceCard from "../components/PlaceCard";

export default class PlaceList extends React.Component {
    constructor(props) {
        super(props);

        this.renderItem = this.renderItem.bind(this);
    }

    static get propTypes() {
        return {
            data: PropTypes.array,
            handlePress: PropTypes.func,
            userPlaces: PropTypes.array,
        };
    }

    renderItem = ({ item, index }) => {
        const isVisited =
            this.props.userPlaces &&
            utilities.isValueInArray(item.id, this.props.userPlaces);

        return (
            <PlaceCard
                place={item}
                handlePress={() => this.props.handlePress(item)}
                isVisited={isVisited}
            />
        );
    };

    render() {
        return (
            <FlatList
                keyExtractor={item => item.id}
                data={this.props.data}
                renderItem={this.renderItem}
                style={styles.wrapper}
                contentContainerStyle={styles.container}
                extraDate={this.props.userLocation}
            />
        );
    }
}

const styles = StyleSheet.create({
    wrapper: {
        alignSelf: "stretch",
    },
    container: {
        paddingTop: 16,
        paddingBottom: 56 + 16,
    },
});
