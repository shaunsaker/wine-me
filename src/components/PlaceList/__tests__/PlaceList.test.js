import React from 'react';
import renderer from 'react-test-renderer';

import PlaceList from '..';
import PLACES from '../../../mockData/PLACES';
import utils from '../../../utils';

jest.mock('../PlaceCard', () => 'PlaceCard');

describe('PlaceList', () => {
  const placesArray = utils.objects.convertObjectToArray(PLACES);

  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<PlaceList data={placesArray} />);

      expect(component).toMatchSnapshot();
    });

    // NOTE: We don't test render methods (those are handle in the test(s) above)
  });
});
