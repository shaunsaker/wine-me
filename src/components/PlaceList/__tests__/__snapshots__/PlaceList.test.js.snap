// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceList renders renders with minimum required props 1`] = `
<RCTScrollView
  contentContainerStyle={Object {}}
  data={
    Array [
      Object {
        "address": "Stellenbosch, South Africa",
        "id": "1",
        "location": Object {
          "lat": -33.9263749,
          "lng": 18.9319848,
        },
        "name": "Bartinney Tasting Shed",
        "openingHours": Array [
          "Monday: 12:00 – 5:00 PM",
          "Tuesday: 12:00 – 5:00 PM",
          "Wednesday: 12:00 – 5:00 PM",
          "Thursday: 12:00 – 5:00 PM",
          "Friday: 12:00 – 8:00 PM",
          "Saturday: 10:00 AM – 12:30 PM",
          "Sunday: Closed",
        ],
        "phoneNumber": "021 885 1013",
        "photoReferences": Array [
          "CmRaAAAAF5bkjtj7bFEGaZqTN1_uZrJMvM6EEOFle3UIDd5JEbZwb281f2F4EH-XR1o6frihMiOzrJEx1KlczmV8o4qLrYygDZE-TKthSRT1SdzvrEQ6AjQbCG6sVeZBR48V0lMLEhDkpyqJh9O9z4Ai-7All7swGhQwTkHxJ77yIZ9d0c8oqWifUrL6eA",
        ],
        "website": "http://www.bartinney.co.za",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={3}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={Object {}}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "paddingBottom": 10,
            "paddingHorizontal": 15,
          }
        }
      >
        <PlaceCard
          place={
            Object {
              "address": "Stellenbosch, South Africa",
              "id": "1",
              "location": Object {
                "lat": -33.9263749,
                "lng": 18.9319848,
              },
              "name": "Bartinney Tasting Shed",
              "openingHours": Array [
                "Monday: 12:00 – 5:00 PM",
                "Tuesday: 12:00 – 5:00 PM",
                "Wednesday: 12:00 – 5:00 PM",
                "Thursday: 12:00 – 5:00 PM",
                "Friday: 12:00 – 8:00 PM",
                "Saturday: 10:00 AM – 12:30 PM",
                "Sunday: Closed",
              ],
              "phoneNumber": "021 885 1013",
              "photoReferences": Array [
                "CmRaAAAAF5bkjtj7bFEGaZqTN1_uZrJMvM6EEOFle3UIDd5JEbZwb281f2F4EH-XR1o6frihMiOzrJEx1KlczmV8o4qLrYygDZE-TKthSRT1SdzvrEQ6AjQbCG6sVeZBR48V0lMLEhDkpyqJh9O9z4Ai-7All7swGhQwTkHxJ77yIZ9d0c8oqWifUrL6eA",
              ],
              "website": "http://www.bartinney.co.za",
            }
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
